README.md

```diff
@@ -121,9 +125,13 @@ some_ros_package
 - Once the world has launched, it is common for the robot to be initially unable to move. Just lift it up a bit in gazebo and drop it for this to be fixed
 - To manually control the robot, run `rosrun turtlesim turtle_teleop_key /turtle1/cmd_vel:=/robot/cmd_vel`
 
-## Github Procedure
-- We follow the "Feature Branch Workflow"
-- A good tutorial can be found [here](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow)
+## Gazebo Testing Procedure
+1. Ask the node creator for a generally specification as to how the node should behave
+- Try and come up with awkward situations which could break the node or cause it to misbehave
+- Test the specification provided by the original node creator, plus any awkward situations you came up with, in Gazebo
+- If the node does not behave as expected (for example, produces the incorrect output) then add unit test(s) to the node (there should already be a testing file present) that mimics the situation(s) that failed in Gazebo (so they should fail if you run the unit tests). **All** Unit Tests should be on a new branch named `NODE-NAME-gazebo-testing`
```

Comment:
Should we consider creating a rosbag to store what caused the crash and replay it for the rostest?